app/uk/gov/hmrc/individualsifapistub/domain/JsonFormatters.scala

```diff
@@ -17,17 +17,288 @@
 package uk.gov.hmrc.individualsifapistub.domain
 
 import org.joda.time.DateTime
+import play.api.libs.functional.syntax.{unlift, _}
 import play.api.libs.json.JodaWrites._
+import play.api.libs.json.Reads.{maxLength, minLength, pattern}
 import play.api.libs.json._
+import uk.gov.hmrc.individualsifapistub.domain.PayeResponseObject._
+import uk.gov.hmrc.individualsifapistub.domain.SaResponseObject.{dateStringPattern, paymentAmountValidator, taxYearPattern, utrPattern}
 
 import scala.util.{Failure, Try}
 
 object JsonFormatters {
   implicit val dateFormatDefault = new Format[DateTime] {
     override def reads(json: JsValue): JsResult[DateTime] = JodaReads.DefaultJodaDateTimeReads.reads(json)
+
     override def writes(o: DateTime): JsValue = JodaDateTimeNumberWrites.writes(o)
   }
 
+  implicit val addressFormat: Format[Address] = Format(
```

Comment:
We need to group these in the files of the case classes they're associated with - there are multiple Address formats with different validation rules (as per the schema), and we don't want them conflicting.